src/qt/listctrl.cpp

```diff
@@ -980,9 +1039,146 @@ class wxQtListTreeWidget : public wxQtEventSignalHandler< QTreeView, wxListCtrl
         return header() != nullptr ? header()->height() : 0;
     }
 
+    int GetRowCount() const
+    {
+        if ( model() )
+            return model()->rowCount();
+
+        return 0;
+    }
+
+    int GetCountPerPage() const
+    {
+        // this may not be exact but should be a good approximation:
+        const int h = rowHeight(model()->index(0, 0));
+        if ( h )
+            return viewport()->height() / h;
+        else
+            return 0;
+    }
+
+protected:
+    virtual void currentChanged(const QModelIndex& current,
+                                const QModelIndex& previous) override
+    {
+        EmitListEvent(wxEVT_LIST_ITEM_FOCUSED, current);
+        QTreeView::currentChanged(current, previous);
+    }
+
+    virtual void selectionChanged(const QItemSelection &selected,
+                                  const QItemSelection &deselected) override;
+
+    // Event handlers
+    virtual void mousePressEvent(QMouseEvent* event) override
+    {
+        const QModelIndex itemPressed = indexAt(event->pos());
+
+        BaseClass::mousePressEvent(event);
+
+        if ( !itemPressed.isValid() )
+        {
+            // wx: deselect all if clicking on empty space
```

Comment:
I don't think it's a requirement of wx API, so if Qt applications don't do this, e.g. if clicking on an empty space doesn't change the selection in standard KDE applications, I think it would make more sense not to do it in the wxQt ones neither (and updated/exclude the tests relying on this happening if we have any).